### PR TITLE
chore(orchestration): idempotent

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
What:
`idempotent` flag on workflows.
If set to true, before executing the workflow, it tries to load from memory/Redis and then from Postgres. If available the same transaction is returned making the flow idempotent.
The `idempotency-key` is the transaction id itself.

### CODE REVIEW

File: packages/core/orchestration/src/transaction/transaction-orchestrator.ts [hunk 1]
```diff
@@ -798,7 +814,7 @@ export class TransactionOrchestrator extends EventEmitter {
       flow.state = TransactionState.INVOKING
       flow.startedAt = Date.now()
 
-      if (this.options?.store) {
+      if (this.getOptions().store) {
```
Comment (riqwan) [inline packages/core/orchestration/src/transaction/transaction-orchestrator.ts:817]: is store guaranteed here?

```suggestion
      if (this.getOptions()?.store) {
```

**Review by riqwan — APPROVED** [review]
clean idea! nice 🎉

File: packages/core/orchestration/src/transaction/types.ts [hunk 2]
```diff
@@ -118,6 +118,11 @@ export type TransactionModelOptions = {
    */
   storeExecution?: boolean
 
+  /**
+   * If true, the workflow will be executed only once with the same transaction id.
```
Comment (sradevski) [inline packages/core/orchestration/src/transaction/types.ts]:
```suggestion
   * If true, the workflow will use the transaction ID as the key to ensure only-once execution
```

**Review by sradevski — APPROVED** [review]
LGTM 👍

File: packages/core/orchestration/src/transaction/transaction-orchestrator.ts (hunk 1 above)
Comment (riqwan) [reply packages/core/orchestration/src/transaction/transaction-orchestrator.ts:817]: ignore this. I read it wrong :D

**Pull request merged**